### PR TITLE
Introduce Controls

### LINKED ISSUE #79: Redundant temperature entities select and sensor
For my washing machine, there are two redundant entities for temperature - one select and one sensor entity.
image

### PR DESCRIPTION
**Pull request by rowysock**
- Controls provide additional level of abstraction between HomeWhiz device config and HA entities
- Controls are HA independent objects describing device read/write capabilities.
- Main idea is to convert device configuration to controls in a single place to avoid code duplication and fragmentation.
- Controls are easier to test than HA Entities
- Controls will be helpful when extracting HomeWhiz related logic to separate PIP package

Closes #79

### CODE REVIEW

File: custom_components/homewhiz/translations/select.de.json [hunk 1]
```diff
@@ -55,28 +55,11 @@
...
       "SPIN_1400": "1400"
     },
-    "homewhiz__WASHER_PREWASH": {
-      "PREWASH_OFF": "Aus",
-      "PREWASH_ON": "Ein"
-    },
-    "homewhiz__WASHER_EXTRARINSE": {
-      "EXTRA_RINSE_OFF": "Nachsp\u00fclen aus",
-      "EXTRA_RINSE_ON": "Nachsp\u00fclen ein"
-    },
```
Comment (TechHummel) [inline custom_components/homewhiz/translations/select.de.json:65]: I'm not sure why this is removed. No other entries in the translation files contain this information / translation.

File: custom_components/homewhiz/sensor.py [hunk 2]
```diff
@@ -1,204 +1,71 @@
...
-        self._attr_device_info = build_device_info(unique_name, data)
-
-        self._localization = data.contents.localization
-        self.entity_description = description
-        self._value_fn = description.value_fn
+        super().__init__(coordinator, device_name, control.key, data)
+        self._control = control
+        if isinstance(control, TimeControl):
+            self._attr_icon = "mdi:clock-outline"
+            self._attr_native_unit_of_measurement = "min"
```
Comment (TechHummel) [inline custom_components/homewhiz/sensor.py:40]: Should be maybe use units from const?
https://github.com/home-assistant/core/blob/dev/homeassistant/const.py#L592

**Review by TechHummel — APPROVED** [review]

**Comment by TechHummel** [comment]
I will give this a try over the next couple of days, but you can already merge if you want.

File: custom_components/homewhiz/translations/select.de.json (hunk 1 above)
Comment (rowysock) [reply custom_components/homewhiz/translations/select.de.json:65]: Replaced with HA switches. No need for translations any longer

File: custom_components/homewhiz/sensor.py (hunk 2 above)
Comment (rowysock) [reply custom_components/homewhiz/sensor.py:40]: Good idea, but it has not yet been released: https://github.com/home-assistant/core/commit/6295cf3dc4d017e6f3151c2b8014e26bada78343

**Comment by rowysock** [comment]
Tested it myself AC and Oven via Wi-Fi as well as Washing machine via bluetooth. Works well. I will merge it

**Pull request merged**